server/src/main/java/io/crate/metadata/information/InformationColumnsTableInfo.java

```diff
@@ -113,6 +114,11 @@
             .add("udt_name", STRING, ignored -> null)
             .add("check_references", STRING, ignored -> null)
             .add("check_action", INTEGER, ignored -> null)
+            .startObject("object_details")
+                .add("name", STRING , r -> r.info.column().name())
+                .add("leaf", STRING , r -> r.info.column().leafName())
+                .add("path", STRING_ARRAY, r -> r.info.column().path())
+            .endObject()
```

Comment:
@seut @jweberpm  any inputs here on the naming?

Reply:
I'm not sure about the top column name `object_details` as I'm if its then obvious enough. Maybe a bit more verbose like `object_column_details` or `object_type_details`? Don't like my suggestions much but can't think of better ones atm.

Reply:
I think `object_type_details` captures the content better, `object_details` might be a little missleading.

Reply:
within a single row, we are referring either to a column or a single property and not necessarily an object.

maybe ... 
`property_root TEXT`
`property_name TEXT`
`property_path TEXT`
all nullable ?

or 

`property OBJECT` with `root`, `name`, `path`

---

additionally one might add
`type` (which could be column or property)
or `is_property` (boolean)

Reply:
> within a single row, we are referring either to a column or a single property and not necessarily an object.

I think it should only show values on columns that are of type object, and on all other columns it should be null

Reply:
Maybe like this?

```
column_details (
  name text,
  path text[]
)
```

I'm also tending to remove `leaf`. What are the reasons to include it?
I'm not sure if it is common enough to retrieve it to warrant special casing it (and it can be resolved via path + array functions)

Reply:
@mfussenegger you are right, having the path should be sufficient. The idea was to query for distinct (e.g. 1st level) properties. This is better to be done with the path anyways.

e.g. to retrieve all first level properties for an object column:

```sql
SELECT column_details[1]['path'] FROM information_schema.columns
where
  table_name = 'table_partitions'
  AND table_schema = 'information_schema'
  AND column_details['name'] = 'settings'
  and array_length(column_details['path'],1) = 1
limit 100;
```

Reply:
@seut @jweberpm 

how about `column_details`, `details` or `structure`?

Reply:
+1 for `column_details`

Reply:
Column_details sounds good.